File: frontend/src/types/knowledge.ts
export interface KnowledgeEntry {
  id: string;
  title: string;
  content: string;
  source_type: string;
  source_id: string | null;
  tags: string[] | null;
  created_at: string;
  updated_at: string;
}

export interface KnowledgeCreate {
  title: string;
  content: string;
  source_type: string;
  source_id?: string;
  tags?: string[];
}

export interface KnowledgeSearchResult {
  entry: KnowledgeEntry;
  similarity: number;
}

File: api/schemas/knowledge.py
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field

VALID_SOURCE_TYPES = ["agent", "analyst", "rule", "conversation"]


class KnowledgeCreate(BaseModel):
    title: str = Field(..., max_length=300)
    content: str
    source_type: str = Field(..., description="One of: " + ", ".join(VALID_SOURCE_TYPES))
    source_id: Optional[uuid.UUID] = None
    tags: Optional[list[str]] = None


class KnowledgeResponse(BaseModel):
    id: uuid.UUID
    title: str
    content: str
    source_type: str
    source_id: Optional[uuid.UUID]
    tags: Optional[list[str]]
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class KnowledgeSearchResult(BaseModel):
    entry: KnowledgeResponse
    similarity: float
